File: src/main/java/me/lokka30/levelledmobs/commands/subcommands/ReloadSubcommand.java
/*
 * Copyright (c) 2020-2021  lokka30. Use of this source code is governed by the GNU AGPL v3.0 license that can be found in the LICENSE.md file.
 */

package me.lokka30.levelledmobs.commands.subcommands;

import java.util.Collections;
import java.util.List;
import me.lokka30.levelledmobs.LevelledMobs;
import me.lokka30.levelledmobs.misc.FileLoader;
import org.bukkit.command.CommandSender;
import org.bukkit.entity.Player;
import org.jetbrains.annotations.NotNull;

/**
 * Reloads all LevelledMobs configuration from disk
 *
 * @author lokka30
 * @since 2.0
 */
public class ReloadSubcommand implements Subcommand {

    @Override
    public void parseSubcommand(final LevelledMobs main, final @NotNull CommandSender sender,
        final String label, final String[] args) {
        if (!sender.hasPermission("levelledmobs.command.reload")) {
            main.configUtils.sendNoPermissionMsg(sender);
            return;
        }

        main.reloadLM(sender);

        if (main.companion.getHadRulesLoadError() && sender instanceof Player) {
            sender.sendMessage(FileLoader.getFileLoadErrorMessage());
        }
    }

    @Override
    public List<String> parseTabCompletions(final LevelledMobs main, final CommandSender sender,
        final String[] args) {
        return Collections.emptyList(); //No tab completions.
    }
}

File: levelledmobs-plugin-bukkit/src/main/kotlin/io/github/arcaneplugins/levelledmobs/plugin/bukkit/command/ReloadSubCommand.kt
package io.github.arcaneplugins.levelledmobs.plugin.bukkit.command

import io.github.arcaneplugins.levelledmobs.plugin.bukkit.LevelledMobs
import java.util.Collections
import org.bukkit.command.Command
import org.bukkit.command.CommandExecutor
import org.bukkit.command.CommandSender
import org.bukkit.command.TabCompleter

class ReloadSubCommand(private val commands: Commands) : CommandExecutor, TabCompleter {
    override fun onCommand(
        sender: CommandSender,
        command: Command,
        label: String,
        args: Array<out String>?
    ): Boolean {
        if (!commands.checkPermission(sender, "levelledmobs.command.reload")){
            return true
        }

        LevelledMobs.lmInstance.reload()

        return true
    }

    override fun onTabComplete(
        sender: CommandSender,
        command: Command,
        label: String,
        args: Array<out String>?
    ): List<String> {
        return Collections.emptyList()
    }
}